mobile/src/components/courses/courses.tsx

```diff
@@ -1,11 +1,63 @@
-import React, { FC } from 'react';
+import React, { FC, ReactElement, useEffect, useState } from 'react';
+import { FlatList, RefreshControl, View } from 'react-native';
 
-import { Text, View } from '~/components/common/common';
+import { AppColor, DataStatus } from '~/common/enums/enums';
+import { Spinner } from '~/components/common/common';
+import { CourseCard } from '~/components/courses/components/components';
+import { useAppDispatch, useAppSelector } from '~/hooks/hooks';
+import { coursesActions } from '~/store/actions';
+
+import { styles } from './styles';
+
+const Courses: FC = (): ReactElement => {
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
+  const { courses, dataStatus } = useAppSelector((state) => state.courses);
+
+  const handleCoursesLoad = async (): Promise<void> => {
+    await dispatch(
+      coursesActions.getCourses({ title: '', categoryKey: '' }),
+    ).unwrap();
+  };
```

Comment:
We should only use this when we need to navigate somewhere if request is successful, because we can't navigate inside action 
```suggestion
  const handleCoursesLoad = (): void => {
    dispatch(
      coursesActions.getCourses({ title: '', categoryKey: '' })
    )
  };
```